zerver/models/users.py

```diff
@@ -965,6 +966,8 @@ def get_user_profile_by_id(user_profile_id: int) -> UserProfile:
         "realm",
         "realm__can_access_all_users_group",
         "realm__can_access_all_users_group__named_user_group",
+        "realm__can_move_messages_between_channels_group",
+        "realm__can_move_messages_between_channels_group__named_user_group",
```

Comment:
Do we need this to be prefetched here? This might need some tweaks following #31789.